Drivers/Src/MotorController.c

```diff
@@ -45,13 +49,14 @@ void MotorController_Drive(float newVelocity, float motorCurrent){
  * @param   message the buffer in which the info for the CAN message will be stored
  * @return  SUCCESS if a message is read
  */ 
-ErrorStatus MotorController_Read(CANbuff *message, void *p_arg){
-    car_state_t *car_state = (car_state_t *) p_arg;
+ErrorStatus MotorController_Read(CANbuff *message, car_state_t *car){
+    car_state_t *car_state = (car_state_t *) car;
```

Comment:
Since you're passing in a car_state_t pointer now, there's no reason to cast it, you can just use it directly.